bottom/protocol.py

```diff
@@ -1,29 +1,38 @@
 import asyncio
 import logging
 from bottom.unpack import unpack_command
+from typing import Optional, TYPE_CHECKING
+
+if TYPE_CHECKING:
```

Comment:
`typing.TYPE_CHECKING` is always `False` at runtime, so there isn't actually any circular import. This flag is seen as `True` by mypy (and other typecheckers) and as we can see it handles this case pretty well. There has to be some kind of import for mypy to follow; otherwise it wouldn't know what `Client` actually is (well, in this codebase there is only one `Client`, but I hope you get the point). It's fine.